From aa2668c682d17664526069231104d977f0b9a9d0 Mon Sep 17 00:00:00 2001
From: Sachin Saxena <sachin.saxena@nxp.com>
Date: Wed, 4 Mar 2026 14:37:22 +0530
Subject: [PATCH 1/2] add support for cross compilation

    export BUILD_DIR=build-aarch64
    rm -rf build-aarch64/;mkdir build-aarch64
    cmake -S . -B "$BUILD_DIR" -G "Unix Makefiles"  -DCMAKE_TOOLCHAIN_FILE=$PWD/aarch64-none-linux.cmake -DHOLOLINK_BUILD_ONLY_NATIVE=ON -DHOLOLINK_BUILD_PYTHON=OFF -DHOLOLINK_BUILD_EXAMPLES=ON -DHOLOLINK_BUILD_TESTS=ON -DHOLOLINK_BUILD_TOOLS=OFF
    make -C "$BUILD_DIR" -j"$(nproc)"

    For DPDK add HSB_DPDK_BUILD=ON

Signed-off-by: Hemant Agrawal <hemant.agrawal@nxp.com>
Upstream-Status: Inappropriate [i.MX95 board BSP specific]
---
 aarch64-none-linux.cmake | 43 ++++++++++++++++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)
 create mode 100644 aarch64-none-linux.cmake

diff --git a/aarch64-none-linux.cmake b/aarch64-none-linux.cmake
new file mode 100644
index 0000000..747dd32
--- /dev/null
+++ b/aarch64-none-linux.cmake
@@ -0,0 +1,43 @@
+# Copyright 2026 NXP
+
+# Target system
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_PROCESSOR aarch64)
+
+if(NOT DEFINED ARM_TOOLCHAIN_ROOT AND DEFINED ENV{ARM_TOOLCHAIN_ROOT})
+  set(ARM_TOOLCHAIN_ROOT "$ENV{ARM_TOOLCHAIN_ROOT}")
+endif()
+
+if(NOT ARM_TOOLCHAIN_ROOT)
+  message(FATAL_ERROR "ARM_TOOLCHAIN_ROOT is not set. Pass -DARM_TOOLCHAIN_ROOT=... or export ARM_TOOLCHAIN_ROOT.")
+endif()
+
+
+# Compilers (note: 'aarch64-none-linux-gnu-*' names)
+set(CMAKE_C_COMPILER   "${ARM_TOOLCHAIN_ROOT}/bin/aarch64-none-linux-gnu-gcc")
+set(CMAKE_CXX_COMPILER "${ARM_TOOLCHAIN_ROOT}/bin/aarch64-none-linux-gnu-g++")
+
+
+set(CMAKE_C_FLAGS_RELEASE   "${CMAKE_C_FLAGS_RELEASE} -O3 -mcpu=cortex-a55+crc")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -mcpu=cortex-a55+crc")
+
+# Optional: ar/ranlib/strip for consistency
+set(CMAKE_AR      "${ARM_TOOLCHAIN_ROOT}/bin/aarch64-none-linux-gnu-ar")
+set(CMAKE_RANLIB  "${ARM_TOOLCHAIN_ROOT}/bin/aarch64-none-linux-gnu-ranlib")
+set(CMAKE_STRIP   "${ARM_TOOLCHAIN_ROOT}/bin/aarch64-none-linux-gnu-strip")
+
+# Sysroot so CMake finds target headers/libs (libc, zlib, etc.)
+# This path exists in ARM GNU toolchain layouts:
+set(CMAKE_SYSROOT "${ARM_TOOLCHAIN_ROOT}/aarch64-none-linux-gnu/libc")
+
+# Ensure CMake searches inside the sysroot for target libs/includes/packages
+set(CMAKE_FIND_ROOT_PATH "${CMAKE_SYSROOT}")
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+
+# Enforce C++17 (needed for std::variant)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
-- 
2.34.1

